pkg/operation/botanist/extensions/operatingsystemconfig/operatingsystemconfig.go

```diff
@@ -0,0 +1,611 @@
+	var fns []flow.TaskFn
+
+	_ = o.forEachWorkerPoolAndPurpose(context.Background(), func(_ context.Context, worker gardencorev1beta1.Worker, purpose extensionsv1alpha1.OperatingSystemConfigPurpose) error {
+		fns = append(fns, func(ctx context.Context) error {
+			return fn(ctx, worker, purpose)
+		})
+		return nil
+	})
+
+	return fns
+}
+
+// AppendCABundle sets the CABundle value.
+func (o *operatingSystemConfig) SetCABundle(val *string) {
+	o.values.CABundle = val
+}
+
+// SetKubeletCACertificate sets the KubeletCACertificate value.
+func (o *operatingSystemConfig) SetKubeletCACertificate(cert string) {
+	o.values.KubeletCACertificate = cert
+}
+
+// SetSSHPublicKey sets the SSHPublicKey value.
+func (o *operatingSystemConfig) SetSSHPublicKey(key string) {
+	o.values.SSHPublicKey = key
+}
+
+// WorkerNameToOperatingSystemConfigsMap returns a map whose key is a worker name and whose value is a structure
+// containing both the downloader as well as the original operating system config data.
+func (o *operatingSystemConfig) WorkerNameToOperatingSystemConfigsMap() map[string]*OperatingSystemConfigs {
+	return o.workerNameToOSCs
+}
+
+func (o *operatingSystemConfig) newDeployer(worker gardencorev1beta1.Worker, purpose extensionsv1alpha1.OperatingSystemConfigPurpose) deployer {
+	criName := extensionsv1alpha1.CRINameDocker
+	if worker.CRI != nil {
+		criName = extensionsv1alpha1.CRIName(worker.CRI.Name)
+	}
+
+	caBundle := o.values.CABundle
+	if worker.CABundle != nil {
+		if caBundle == nil {
+			caBundle = worker.CABundle
+		} else {
+			*caBundle = fmt.Sprintf("%s\n%s", *caBundle, *worker.CABundle)
+		}
+	}
+
+	kubeletConfigParameters := o.values.KubeletConfigParameters
+	kubeletCLIFlags := o.values.KubeletCLIFlags
+	if worker.Kubernetes != nil && worker.Kubernetes.Kubelet != nil {
+		kubeletConfigParameters = components.KubeletConfigParametersFromCoreV1beta1KubeletConfig(worker.Kubernetes.Kubelet)
+		kubeletCLIFlags = components.KubeletCLIFlagsFromCoreV1beta1KubeletConfig(worker.Kubernetes.Kubelet)
+	}
+	setDefaultEvictionMemoryAvailable(kubeletConfigParameters.EvictionHard, kubeletConfigParameters.EvictionSoft, o.values.MachineTypes, worker.Machine.Type)
+
+	return deployer{
+		client:                  o.client,
+		worker:                  worker,
+		purpose:                 purpose,
+		key:                     Key(worker.Name, o.values.KubernetesVersion),
+		namespace:               o.values.Namespace,
+		apiServerURL:            o.values.APIServerURL,
+		caBundle:                caBundle,
+		clusterDNSAddress:       o.values.ClusterDNSAddress,
+		clusterDomain:           o.values.ClusterDomain,
+		criName:                 criName,
+		images:                  o.values.Images,
+		kubeletCACertificate:    o.values.KubeletCACertificate,
+		kubeletConfigParameters: kubeletConfigParameters,
```

Comment:
```suggestion
		globalKubeletConfigParameters: kubeletConfigParameters,
```

Makes it a bit more clear?

Reply:
Not sure, I would like to keep it similar to the `kubeletCLIFlags` field.